### PR TITLE
Fix/plf 8320

### PR DESCRIPTION
**Pull request by AzmiTouil**

### CODE REVIEW

File: component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/MigrationTest.java [hunk 1]
```diff
@@ -122,9 +122,7 @@ public void setUp() throws Exception {
     identityMigrationService = getService(IdentityMigrationService.class);
     activityMigration = getService(ActivityMigrationService.class);
     spaceMigrationService = getService(SpaceMigrationService.class);
-
-    deleteIdentities();
```
Comment (boubaker) [inline component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/MigrationTest.java]: Why deleting this instruction ?
IMO, the identities deletion should be done before executing tests. You may encounter a problem if it's not encapsulated inside `begin();` and `end();`

File: component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/AsynMigrateActivitiesTest.java [hunk 2]
```diff
@@ -36,12 +36,18 @@
   @MaxQueryNumber(36990)
   public void testMigrationActivities() throws Exception {
     // create jcr data
+
+    ProviderRootEntity providerEntity = identityJCRStorage.getProviderRoot();
+    ProviderEntity organization = providerEntity.getProvider(OrganizationIdentityProvider.NAME);
```
Comment (boubaker) [inline component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/AsynMigrateActivitiesTest.java]: Why adding this instruction here ? It's not needed for test. If you need to start a session, it should be made by starting a transaction

File: component/common/src/main/java/org/exoplatform/social/common/lifecycle/SocialChromatticLifeCycle.java [hunk 3]
```diff
@@ -37,8 +36,8 @@
   public static final String SOCIAL_LIFECYCLE_NAME = "soc";
 
   public ChromatticSession getSession() {
-    if (invalidSession()) {
-      reCreateSession();
+    if(session.get() == null){
```
Comment (boubaker) [inline component/common/src/main/java/org/exoplatform/social/common/lifecycle/SocialChromatticLifeCycle.java]: we shouldn't create a session manually here, it should be created automatically by `RequestLifecycle.begin`

File: component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/MigrationTest.java (hunk 1 above)
Comment (AzmiTouil) [reply component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/MigrationTest.java]: done

File: component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/AsynMigrateActivitiesTest.java (hunk 2 above)
Comment (AzmiTouil) [reply component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/concurrency/AsynMigrateActivitiesTest.java]: Okay
done

File: component/common/src/main/java/org/exoplatform/social/common/lifecycle/SocialChromatticLifeCycle.java (hunk 3 above)
Comment (AzmiTouil) [reply component/common/src/main/java/org/exoplatform/social/common/lifecycle/SocialChromatticLifeCycle.java]: Always I get a null session, I don't know from where comes the end of the session.

Comment (AzmiTouil) [reply component/common/src/main/java/org/exoplatform/social/common/lifecycle/SocialChromatticLifeCycle.java]: The lifecycle is created but the session is not by default. Affecting the JCR session is done by calling the function onOpenSession() (Before it was done in the reCreateSession function which caused the session leak).
Here we just check if the session is already created , if not we create it by calling onOpenSession(), then whenever there is a call to the getSession() in the same lifecycle, the session won't be recreated and we should get the one that we already set.

**Pull request merged**